kong/db/init.lua

```diff
@@ -739,30 +740,50 @@ do
                               self.strategy, mig.name)
         end
 
+        if mig.name == "000_base" then
+          local is_0_14, err_0_14, is_gt_0_14, missing_0_14_id, missing_0_14_migration =
+            self.connector:is_0_14()
+
+          if err_0_14 then
+            return nil, fmt_err(self, "unable to detect database version (%s) ", err_0_14)
+          end
+
+          if not is_0_14 and not is_gt_0_14 then
+            if missing_0_14_migration then
+              return nil, fmt_err(self, "migration '%s' can only be executed on " ..
+                                  "a %s database that is migrated to version 0.14.x, " ..
+                                  "but database is missing '%s' migration for '%s'",
+                                  mig.name, self.strategy, missing_0_14_migration,
+                                  missing_0_14_id)
+            else
+              return nil, fmt_err(self, "migration '%s' can only be executed on " ..
+                                  "a %s database that is migrated to version 0.14.x, " ..
+                                  "but database is missing '%s'", mig.name, self.strategy,
+                                  missing_0_14_id)
```

Comment:
I feel like both of these error messages can be improved. They should start by explaining what they are trying to do (migrate to 1.0), and should have some sort of closure, to explain to the user which conclusion our heuristics took:
```
Migration to 1.0 can only be executed from a fresh or 0.14.x %s [database/keyspace], but the %s [database/keyspace] is missing some migrations (%s), indicating an older version. Migrate to 0.14.x first.
```

Where the first `%s` is "Postgres/Cassandra", the second one is the name of the database/keyspace itself, and the third one is the name of the migration detected as missing.

The message may be refined, of course. Ultimately, I don't even think we need to separate both possible error cases, which would result in a single, identical error message for both "missing component" and "missing migration(s) in component". Unless we do for another reason>